app/src/main/java/com/example/themealsapp/utils/NetworkState.kt

```diff
@@ -0,0 +1,17 @@
+package com.example.themealsapp.utils
+
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import javax.inject.Inject
+
+class NetworkState @Inject constructor(
+    private val connectivityManager: ConnectivityManager
+) {
+
+
+    fun isInternetOn() =
```

Comment:
is this fully working?

Reply:
I will say I have seen the app function as intended with the project code in its current state - we should be good to go

Reply:
it does